### PR TITLE
Add Container tests and fix a debug assertion

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

**Comment by hughbe** [comment]
Test OSX x64 Debug Build
Test NETFX x86 Release Build

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Container.cs [hunk 1]
```diff
@@ -265,60 +225,67 @@ protected virtual void ValidateName(IComponent component, string name)
...
             }
 
-            if (name != null)
+            if (name == null)
             {
-                for (int i = 0; i < Math.Min(_siteCount, _sites.Length); i++)
-                {
-                    ISite s = _sites[i];
+                return;
+            }
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Container.cs]: I don't see this as a worthwhile change.  Some folks prefer it one way, other folks prefer it the other.

Comment (danmosemsft) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Container.cs]: Early-out like this helps reduce nesting, which can make code easier to reason about. I guess there is not consensus about that though.

Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Container.cs]: Should I revert this change or not?

Comment (hughbe) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Container.cs]: I reverted this anyway

**Comment by hughbe** [comment]
Test Linux x64 Release Build

**Comment by hughbe** [comment]
@safern is there still no way for contributors to run uapaot tests? :(

**Comment by safern** [comment]
> @safern is there still no way for contributors to run uapaot tests? :(

No, sadly there is no way for you guys to run the uapaot tests :(
